starlite/_signature/models/attrs_signature_model.py

```diff
@@ -0,0 +1,343 @@
+    except (ValueError, TypeError):
+        pass
+
+    return parse(value)
+
+
+def _structure_date(value: Any, cls: type[date]) -> date:
+    if isinstance(value, date) and not isinstance(value, datetime):
+        return value
+
+    if isinstance(value, (float, int, Decimal)):
+        return cls.fromtimestamp(float(value))
+
+    dt = _structure_datetime(value=value, cls=datetime)
+    return cls(year=dt.year, month=dt.month, day=dt.day)
+
+
+def _structure_time(value: Any, cls: type[time]) -> time:
+    if isinstance(value, time):
+        return value
+
+    if isinstance(value, str):
+        return cls.fromisoformat(value)
+
+    dt = _structure_datetime(value=value, cls=datetime)
+    return cls(hour=dt.hour, minute=dt.minute, second=dt.second, microsecond=dt.microsecond, tzinfo=dt.tzinfo)
+
+
+def _structure_timedelta(value: Any, cls: type[timedelta]) -> timedelta:
+    if isinstance(value, timedelta):
+        return value
+    if isinstance(value, (float, int, Decimal)):
+        return cls(seconds=int(value))
+    return cls(seconds=timeparse(value))
+
+
+def _structure_decimal(value: Any, cls: type[Decimal]) -> Decimal:
+    return cls(str(value))
+
+
+def _structure_path(value: Any, cls: type[PurePath]) -> PurePath:
+    return cls(str(value))
+
+
+def _structure_uuid(value: Any, cls: type[UUID]) -> UUID:
+    return value if isinstance(value, UUID) else cls(str(value))
+
+
+def _structure_multidict(value: Any, cls: type[MultiDict]) -> MultiDict:
+    return cls(value)
+
+
+def _structure_str(value: Any, cls: type[str]) -> str:
+    # see: https://github.com/python-attrs/cattrs/issues/26#issuecomment-358594015
+    if value is None:
+        raise ValueError
+    return cls(value)
+
+
+hooks: list[tuple[type[Any], Callable[[Any, type[Any]], Any]]] = [
+    (ASGIConnection, _pass_through_structure_hook),
+    (Decimal, _structure_decimal),
+    (ImmutableState, _pass_through_structure_hook),
+    (MultiDict, _structure_multidict),
+    (PurePath, _structure_path),
+    (Request, _pass_through_structure_hook),
+    (State, _pass_through_structure_hook),
+    (UUID, _structure_uuid),
+    (UploadFile, _pass_through_structure_hook),
+    (WebSocket, _pass_through_structure_hook),
+    (date, _structure_date),
+    (datetime, _structure_datetime),
+    (str, _structure_str),
+    (time, _structure_time),
+    (timedelta, _structure_timedelta),
+    *pydantic_hooks,
+]
+
+
+def _create_default_structuring_hooks(
+    converter: cattrs.Converter,
+) -> tuple[Callable, Callable]:
+    """Create scoped default hooks for a given converter.
+
+    Notes:
+        - We are forced to use this pattern because some types cannot be handled by cattrs out of the box. For example,
+            union types, optionals, complex union types etc.
+        - See: https://github.com/python-attrs/cattrs/issues/311
+    Args:
+        converter: A converter instance
+
+    Returns:
+        A tuple of hook handlers
+    """
+
+    @lru_cache(1024)
```

Comment:
Is this OK if `value` isn't hashable, or is that not a possible outcome here?

Reply:
Value is expected to be a string in this case, and cl is always hashable. 

I think it's ok, but there is a small chance I'm wrong. I think we should try this and see if we hit an issue in the wild.